app/main.py

```diff
@@ -1,3 +1,80 @@
+from __future__ import annotations
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(
+            self,
+            km: int | float
+    ) -> None:
+        self.km = km
+
+    @staticmethod
+    def other_type_check(other: Union[Distance, float, int]) -> bool:
+        if not isinstance(other, (Distance, float, int)):
+            raise TypeError("'other' is not 'Distance', 'float', 'int'!!!")
+        return isinstance(other, Distance)
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union[Distance, float, int]) -> Distance:
+        return (
+            Distance(self.km + other.km)
+            if self.other_type_check(other) else
+            Distance(self.km + other)
+        )
```

Comment:
The `__add__` method can be simplified. Instead of using a ternary operator, you can use an if-else statement. This would make the code easier to read. Moreover, you are repeating the same code (`Distance(self.km + other)`) in the `__add__` and `__iadd__` methods. You can avoid this repetition by creating a private method that performs the addition.